app/views/shared/_comment.html.erb

```diff
@@ -1,7 +1,7 @@
-<% commentable ||= comment.commentable %>
+<% commentable ||= comment.commentable.becomes(comment.commentable.class.base_class) %>
 <div class="comment-box">
```

Comment:
This ensures that rails generates a link to, for example, /requests/3/comments rather than /request_subclass/3/comments which fails. I've got no idea why Rails isn't smart enough to perform this conversion itself in cases where the route doesn't exist for the subclass.